src/Tempest/Database/DatabaseFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use Tempest\Database\Drivers\MySqlDriver;
+use Tempest\Database\Drivers\PostgreSqlDriver;
+use Tempest\Database\Drivers\SQLiteDriver;
+
+final class DatabaseFactory
+{
+    public static function make(DatabaseDialect $dialect, array $options): DatabaseConfig
+    {
+        $instance = new self();
+
+        return new DatabaseConfig(
+            match ($dialect) {
```

Comment:
You could run an `array_filter` on the options and pass those to the class? That should leave the default if not added.